detection_rules/rule.py

```diff
@@ -238,6 +238,58 @@ class ThresholdAlertSuppression:
     duration: AlertSuppressionDuration
 
 
+@dataclass(frozen=True)
+class FilterStateStore:
+    store: definitions.StoreType
+
+
+@dataclass(frozen=True)
+class FilterMeta:
+    alias: Optional[Union[str, None]] = None
+    disabled: Optional[bool] = None
+    negate: Optional[bool] = None
+    controlledBy: Optional[str] = None  # identify who owns the filter
+    group: Optional[str] = None  # allows grouping of filters
+    index: Optional[str] = None
+    isMultiIndex: Optional[bool] = None
+    type: Optional[str] = None
+    key: Optional[str] = None
+    params: Optional[str] = None  # Expand to FilterMetaParams when needed
+    value: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class WildcardQuery:
+    case_insensitive: bool
+    value: str
+
+
+@dataclass(frozen=True)
+class Query:
+    wildcard: Optional[Dict[str, WildcardQuery]] = None
+
+
+@dataclass(frozen=True)
+class Filter:
+    meta: FilterMeta
+    query: Optional[Union[Query, Dict[str, Any]]] = None
+    state: Optional[FilterStateStore] = None
+
+    @pre_load
+    def convert_to_state_field(self, data, **kwargs) -> dict:
```

Comment:
I see what you did here for importing for exporting. Makes sense :)

Reply:
You should be able to get simplify and get rid of these 2 methods by using

```
= field(metadata=dict(data_key="$state"))
```

https://github.com/elastic/detection-rules/pull/3505/files#r1546589383 (should be tested)